lms/static/scripts/frontend_apps/config.ts

```diff
@@ -205,28 +205,40 @@ export type Product = {
  * the backend code.
  */
 export type ConfigObject = {
+  // Configuration present in all modes.
   mode: AppMode;
   api: {
     authToken: string;
-    sync: APICallInfo;
-    viaUrl: APICallInfo;
+
+    // Only present in "basic-lti-launch" mode.
+    sync?: APICallInfo;
+    viaUrl?: APICallInfo;
   };
+  dev: boolean;
+  debug?: DebugInfo;
+  product: Product;
+
+  // Only present in "basic-lti-launch" mode.
   canvas: {
-    speedGrader: SpeedGraderConfig;
+    // Only present in Canvas.
+    speedGrader?: SpeedGraderConfig;
   };
   contentBanner?: ContentBannerConfig;
-  dev: boolean;
-  filePicker: FilePickerConfig;
-  grading: GradingConfig;
+  grading?: GradingConfig; // Only present for instructors
   hypothesisClient: ClientConfig;
   rpcServer: {
     allowedOrigins: string[];
   };
   viaUrl: string;
```

Comment:
It will probably make sense in future to put all these LTI-launch fields under their own key, like we do for other modes. That's beyond the scope of these changes though.